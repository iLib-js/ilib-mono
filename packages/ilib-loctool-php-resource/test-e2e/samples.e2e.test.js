const path = require("path");
const { expectFileToMatchSnapshot, LoctoolRunner } = require("@ilib-mono/e2e-test");

describe("samples", () => {
    describe("php-resource", () => {
        const projectPath = path.resolve(__dirname, "..", "samples", "php-resource");

        beforeAll(async () => {
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        it("should produce an extracted XLIFF file", () => {
            const xliffPath = path.resolve(projectPath, "sample-php-resource-extracted.xliff");
            expectFileToMatchSnapshot(xliffPath);
        });
    });
    describe("regex", () => {
        const projectPath = path.resolve(__dirname, "..", "samples", "regex");

        beforeAll(async () => {
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        it("should produce an extracted XLIFF file", () => {
            const xliffPath = path.resolve(projectPath, "sample-regex-extracted.xliff");
            expectFileToMatchSnapshot(xliffPath);
        });
    });
});
