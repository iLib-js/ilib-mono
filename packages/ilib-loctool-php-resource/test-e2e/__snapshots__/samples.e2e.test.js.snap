// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samples php-resource should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="src/t1.js" source-language="en-US" product-name="sample-php-resource">
    <body>
      <trans-unit id="1" resname="Hello" restype="string" datatype="javascript">
        <source>Hello</source>
      </trans-unit>
      <trans-unit id="2" resname="This is a string that has no translation yet" restype="string" datatype="javascript">
        <source>This is a string that has no translation yet</source>
      </trans-unit>
      <trans-unit id="3" resname="Thank you" restype="string" datatype="javascript">
        <source>Thank you</source>
      </trans-unit>
      <trans-unit id="4" resname="id1" restype="string" datatype="javascript">
        <source>This is a test with a unique id</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`samples regex should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="src/js/test.jsx" source-language="en-US" product-name="sample-regex">
    <body>
      <trans-unit id="2" resname="r870312989" restype="string" datatype="javascript">
        <source>"Products"</source>
      </trans-unit>
      <trans-unit id="4" resname="r289616634" restype="string" datatype="javascript">
        <source>"Solutions"</source>
      </trans-unit>
      <trans-unit id="5" resname="r199753261" restype="string" datatype="javascript">
        <source>"Documentation"</source>
      </trans-unit>
      <trans-unit id="6" resname="r1053329949" restype="string" datatype="javascript">
        <source>"About Us"</source>
      </trans-unit>
      <trans-unit id="7" resname="navbar.label.for.developers" restype="string" datatype="javascript">
        <source>"For Developers"</source>
      </trans-unit>
    </body>
  </file>
  <file original="src/smarty/test.tmpl" source-language="en-US" product-name="sample-regex">
    <body>
      <trans-unit id="1" resname="nav_button_label_text" restype="string" datatype="php">
        <source>'Navigation'</source>
      </trans-unit>
      <trans-unit id="3" resname="copyright_notice" restype="string" datatype="php">
        <source>'Copyright 2024 ACME Inc., All rights reserved.'</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;
